astropy/wcs/__init__.py

```diff
@@ -33,6 +33,6 @@ def get_include():
     """
     Get the path to astropy.wcs's C header files.
     """
-    import os
+    from pathlib import Path
 
-    return os.path.join(os.path.dirname(__file__), "include")
+    return str(Path(__file__).parent / "include")
```

Comment:
Can we do string casting later and return a Path here?

Reply:
No I think this is public API

Reply:
yes, and if I understand correctly it's meant to be used in setuptools `Extension`, whose interface is being made compatible with `Path` objects in recent versions only, so we can't change this return type without subtly breaking any downstream packaging env that wouldn't have access to setuptools 72.2 or newer (the "exotic aarchs" tests are an example of that)

Reply:
I do plan to cut a release soon after the PR is merged

Reply:
ok then let's move this one to draft for now. Thanks !

Reply:
whoops, totally my fault, but this conversation was meant for a different thread https://github.com/astropy/astropy/pull/16936#discussion_r1744343515